metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -1136,6 +1139,34 @@ ControllerResult<AlterPartitionResponseData> alterPartition(
         return ControllerResult.of(records, response);
     }
 
+    /**
+     * Validates that a batch of topics will create less than {@value MAX_PARTITIONS_PER_BATCH}. Exceeding this number of topics per batch
+     * has led to out-of-memory exceptions. We use this validation to fail earlier to avoid allocating the memory.
+     * This validation assumes that all the topics in the batch are valid and can be created.
+     *
+     * @param request a batch of topics to create.
+     * @param defaultNumPartitions default number of partitions to assign if unspecified.
+     * @throws PolicyViolationException if total number of partitions exceeds {@value MAX_PARTITIONS_PER_BATCH}.
+     */
+    static void validateEstimatedTotalNumberOfPartitions(CreateTopicsRequestData request, int defaultNumPartitions) {
```

Comment:
can you remove the "estimated" from the function name? since we're calculating the exact number (not an estimate...)